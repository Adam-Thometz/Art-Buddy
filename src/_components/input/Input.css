.Input {
  display: flex;
  flex-direction: column;
  width: 60%;
  margin: 0 auto;
  text-align: left;
}

.Input-label {
  padding: 10px 0;
  font-size: 2rem;
}

.Input-input {
  border-radius: 25px;
  border: 5px solid #17255A;
  height: 37px;
  padding: 10px;
  font-size: 1.5rem;
}