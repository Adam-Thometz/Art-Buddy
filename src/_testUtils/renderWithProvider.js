import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;