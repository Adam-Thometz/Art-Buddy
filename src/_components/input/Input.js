import React, { useState } from 'react';

import './Input.css';

const Input = ({ type = 'text', label, id,  }) => {
  const [input, setInput] = useState('');

  const handleChange = e => {
    setInput(e.target.value)
  }

  return (
    <div className='Input'>
      <label className='Input-label' htmlFor={id}>{label}</label>
      <input
        className='Input-input'
        type={type}
        id={id}
        name={id}
        onChange={handleChange}
        value={input}
      />
    </div>
  );
};

export default Input;