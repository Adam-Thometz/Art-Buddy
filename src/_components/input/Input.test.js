import React from "react";
import renderWithProvider from "../../_testUtils/renderWithProvider";
import { screen } from "@testing-library/react";
import userEvent from '@testing-library/user-event'
import Input from "./Input";

describe('Input component', () => {
  it('renders without crashing', () => {
    renderWithProvider(<Input />);
  });
  
  it('matches snapshot', () => {
    const { asFragment } = renderWithProvider(<Input />);
    expect(asFragment()).toMatchSnapshot();
  });
  
  it('accepts user input', () => {
    renderWithProvider(<Input />);
    const input = screen.getByRole('textbox');
    userEvent.type(input, 'Hello world!');
    expect(screen.getByDisplayValue('Hello world!')).toBeInTheDocument();
  })
});