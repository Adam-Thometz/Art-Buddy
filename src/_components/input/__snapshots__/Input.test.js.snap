// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Input"
  >
    <label
      class="Input-label"
    />
    <input
      class="Input-input"
      type="text"
      value=""
    />
    <div
      class="Input-keyboard"
    >
      <figure
        class="Icon"
      >
        <img
          alt=""
          src="file"
          style="height: 40px; width: 40px;"
        />
        <figcaption
          style="font-size: 1rem;"
        >
          OPEN KEYBOARD
        </figcaption>
      </figure>
    </div>
  </div>
</DocumentFragment>
`;
